### ISSUE TITLE
load.uid & loader.gid config not working

### ISSUE THREAD

**Issue by ScottLiao920** [gramineproject/gramine#613]
## Description of the problem
`gramine-sgx postgres -d` failed bacause of wrong ownership of the data directory. 
The data directory is ownby user with uid=gid=1000, so I provided `loader.uid=1000` and `loader.gid=1000` in my manifest file.

## Steps to reproduce
* install postgresql & gramine (build from master branch)
* `initdb /pathtodatadirectory`  in my case, `/usr/local/pgsql/data`, issued by user with uid=gid=1000
* `gramine-manifest \
        -Dlog_level=debug \
        -Dexecdir=/usr/local/pgsql/bin \
        -Darch_libdir=/lib/x86_64-linux-gnu \
        postgres.manifest.template >postgres.manifest
`
* `gramine-sgx-sign \
        --manifest postgres.manifest \
        --output postgres.manifest.sgx
`
* `gramine-sgx-get-token --output postgres.token --sig postgres.sig`
* `gramine-sgx postgres -D /usr/local/pgsql/data`

## Expected results
start the postgresql server without any fuss
## Actual results
```
Gramine is starting. Parsing TOML manifest file, this may take some time...
debug: Gramine parsed TOML manifest file successfully
debug: Token file: postgres.token
debug: Read token:
debug:     valid:                 0x00000001
debug:     attr.flags:            0x0000000000000006
debug:     attr.xfrm:             0x0000000000000007
debug:     mr_enclave:            458bf8529e46483a6250826cedea2cae416b6462106f3a9876ddeea48d45aed0
debug:     mr_signer:             578c5dfc73d62536dd34a15c383e9aa4924eafbdf0c131ff2d80ef2db01bc35d
debug:     LE cpu_svn:            0e12ffffff8003000000000000000000
debug:     LE isv_prod_id:        20
debug:     LE isv_svn:            05
debug:     LE masked_misc_select: 0x00000000
debug:     LE attr.flags:         0x0000000000000021
debug:     LE attr.xfrm:          0x0000000000000000
debug: Enclave created:
debug:     base:           0x0000000200000000
debug:     size:           0x0000000200000000
debug:     misc_select:    0x00000000
debug:     attr.flags:     0x0000000000000007
debug:     attr.xfrm:      0x0000000000000007
debug:     ssa_frame_size: 4
debug:     isv_prod_id:    0x00000000
debug:     isv_svn:        0x00000000
debug: Adding pages to SGX enclave, this may take some time...
debug: Adding pages to enclave: 0x3fffa6000-0x400000000 [REG:R--] (manifest) measured
debug: Adding pages to enclave: 0x3fff86000-0x3fffa6000 [REG:RW-] (ssa) measured
debug: Adding pages to enclave: 0x3fff82000-0x3fff86000 [TCS:---] (tcs) measured
debug: Adding pages to enclave: 0x3fff7e000-0x3fff82000 [REG:RW-] (tls) measured
debug: Adding pages to enclave: 0x3fff3e000-0x3fff7e000 [REG:RW-] (stack) measured
debug: Adding pages to enclave: 0x3ffefe000-0x3fff3e000 [REG:RW-] (stack) measured
debug: Adding pages to enclave: 0x3ffebe000-0x3ffefe000 [REG:RW-] (stack) measured
debug: Adding pages to enclave: 0x3ffe7e000-0x3ffebe000 [REG:RW-] (stack) measured
debug: Adding pages to enclave: 0x3ffe6e000-0x3ffe7e000 [REG:RW-] (sig_stack) measured
debug: Adding pages to enclave: 0x3ffe5e000-0x3ffe6e000 [REG:RW-] (sig_stack) measured
debug: Adding pages to enclave: 0x3ffe4e000-0x3ffe5e000 [REG:RW-] (sig_stack) measured
debug: Adding pages to enclave: 0x3ffe3e000-0x3ffe4e000 [REG:RW-] (sig_stack) measured
debug: Adding pages to enclave: 0x3ffa1a000-0x3ffa66000 [REG:R-X] (code) measured
debug: Adding pages to enclave: 0x3ffa66000-0x3ffa6a000 [REG:RW-] (data) measured
debug: Adding pages to enclave: 0x3ffa6a000-0x3ffe3e000 [REG:RW-] (bss) measured
debug: Adding pages to enclave: 0x200000000-0x3ffa1a000 [REG:RWX] (free)
debug: Added all pages to SGX enclave
debug: Enclave initializing:
debug:     enclave id:   0x00000003fffff000
debug:     mr_enclave:   458bf8529e46483a6250826cedea2cae416b6462106f3a9876ddeea48d45aed0
-----------------------------------------------------------------------------------------------------------------------
Gramine detected the following insecure configurations:

  - sgx.debug = true                           (this is a debug enclave)
  - loader.log_level = warning|debug|trace|all (verbose log level, may leak information)
  - loader.insecure__use_cmdline_argv = true   (forwarding command-line args from untrusted host to the app)
  - sgx.allowed_files = [ ... ]                (some files are passed through from untrusted host without verification)

Gramine will continue application execution, but this configuration must not be used in production!
-----------------------------------------------------------------------------------------------------------------------

warning: DkVirtualMemoryProtect is unimplemented in Linux-SGX PAL
[::] debug: Gramine was built from commit: 1.1post-UNRELEASED
[::] debug: Host: Linux-SGX
[::] debug: LibOS xsave_enabled 1, xsave_size 0x340(832), xsave_features 0x7
[::] debug: Initial VMA region 0x3ff66f000-0x3ff715000 (LibOS) bookkeeped
[::] debug: Initial VMA region 0x3fffa6000-0x400000000 (manifest) bookkeeped
[::] debug: ASLR top address adjusted to 0x3d472c000
[::] debug: host is Linux-SGX, adding SGX-specific /dev/attestation files: report, quote, etc.
[::] debug: Shim loaded at 0x3ff66f000, ready to initialize
[::] debug: mounting "file:." (chroot) under /
[::] debug: mounting "proc" (pseudo) under /proc
[::] debug: mounting "dev" (pseudo) under /dev
[::] debug: mounting "dev:tty" (chroot) under /dev/tty
[::] debug: mounting "sys" (pseudo) under /sys
[::] debug: Creating pipe: pipe.srv:b1d29dfe5530da3f0e72bfc6878f5cfd1b8284e846b8f1e28e85f95520766028
[P1:T1:] debug: mounting "file:/usr/local/lib/x86_64-linux-gnu/gramine/runtime/glibc" (chroot) under /lib
[P1:T1:] debug: mounting "file:/lib/x86_64-linux-gnu" (chroot) under /lib/x86_64-linux-gnu
[P1:T1:] debug: mounting "file:/usr/local/pgsql/bin/postgres" (chroot) under /usr/local/pgsql/bin/postgres
[P1:T1:] debug: mounting "file:/usr/local/pgsql/" (chroot) under /usr/local/pgsql/
[P1:T1:postgres] debug: Creating pipe: pipe.srv:a97b9e341081d61166a8582713c807108c6da57d2369b18ff2637882ac5a0a7e
[P1:T1:postgres] debug: Allocating stack at 0 (size = 1048576)
[P1:T1:postgres] debug: loading "file:/usr/local/pgsql//bin/postgres"
[P1:T1:postgres] debug: find_interp: searching for interpreter: /lib/ld-linux-x86-64.so.2
[P1:T1:postgres] debug: loading "file:/usr/local/lib/x86_64-linux-gnu/gramine/runtime/glibc/ld-linux-x86-64.so.2"
[P1:T1:postgres] debug: Creating pipe: pipe.srv:1
[P1:T1:postgres] debug: Creating pipe: pipe.srv:1f0c6d7f551ae05a40dfdeaf09cc912d2aa59820da6023866baea47a8ba11adf
[P1:T1:postgres] debug: Shim process initialized
[P1:shim] debug: IPC worker started
[P1:T1:postgres] debug: glibc register library /lib/librt.so.1 loaded at 0x3d3ff1000
[P1:T1:postgres] debug: glibc register library /lib/libdl.so.2 loaded at 0x3d3fec000
[P1:T1:postgres] debug: glibc register library /lib/libm.so.6 loaded at 0x3d3f13000
[P1:T1:postgres] debug: glibc register library /lib/libc.so.6 loaded at 0x3d3d20000
FATAL:  data directory "/usr/local/pgsql/data" has wrong ownership
HINT:  The server must be started by the user that owns the data directory.
[P1:T1:postgres] debug: ---- shim_exit_group (returning 1)
[P1:T1:postgres] debug: clearing POSIX locks for pid 1
[P1:T1:postgres] debug: sync client shutdown: closing handles
[P1:T1:postgres] debug: sync client shutdown: waiting for confirmation
[P1:T1:postgres] debug: sync client shutdown: finished
[P1:shim] debug: IPC worker: exiting worker thread
[P1:T1:postgres] debug: process 1 exited with status 1
debug: DkProcessExit: Returning exit code 1
```
## Additional information
my manifest template:
```
loader.entrypoint = "file:{{ gramine.libos }}"
libos.entrypoint = "{{ execdir }}/postgres"

loader.log_level = "{{ log_level }}"
loader.uid = 1000
loader.gid = 1000
sys.stack.size = "1M"

loader.insecure__use_cmdline_argv = true

loader.env.LD_LIBRARY_PATH = "/lib:{{ arch_libdir }}"
loader.env.PATH = "{{ execdir }}"
# Set HOME to suppress "warning: cannot find home directory; cannot read ~/.sqliterc"
loader.env.HOME = "/"

fs.mounts = [
  { path = "/lib", uri = "file:{{ gramine.runtimedir() }}" },
  { path = "{{ arch_libdir }}", uri = "file:{{ arch_libdir }}" },
  { path = "{{ execdir }}/postgres", uri = "file:{{ execdir }}/postgres" },
  { path = "/usr/local/pgsql/", uri = "file:/usr/local/pgsql/" },
]

sgx.debug = true
sgx.enclave_size = "8G"
sgx.thread_num = 4


sgx.trusted_files = [
  "file:{{ gramine.libos }}",
  "file:{{ execdir }}/postgres",
  "file:{{ gramine.runtimedir() }}/",
  "file:{{ arch_libdir }}/",
  "file:/usr/local/pgsql/bin/",
  "file:/usr/local/pgsql/data/",
  "file:/usr/local/pgsql/lib/",
  "file:/usr/local/pgsql/share/",
  "file:/usr/local/pgsql/include/",
  "file:/usr/local/pgsql/data/postgresql.conf",
]
```

**Comment by dimakuv** [comment]
@ScottLiao920 Could you try the latest Gramine (you can just git-clone and build, as you did previously)? We just released v1.2, so I wonder if this problem got fixed by itself (we had many bug fixes in the FS subsystem recently).

Also, can you show us `ls -lh /usr/local/pgsql/data`? Just to be sure that the ownership of this file is as expected.

**Comment by dimakuv** [comment]
I'm closing this issue since it wasn't active for almost a year already.
